wildmatch_test.go

```diff
@@ -202,7 +202,7 @@ var Cases = []*Case{
 	{
 		Pattern: `**/bar/*`,
 		Subject: `deep/foo/bar/baz/`,
-		Match:   false,
+		Match:   true,
 	},
 	{
```

Comment:
This is not immediately germane to this PR, but I would love to see a few extra tests in here; these are things I had lurking in my local dev copy but have never turned into a PR.  If you feel like including any of them, please do, but it's not urgent.
```go
	{
		Pattern: `**/bar/**`,
		Subject: `deep/foo/bar`,
		Match:   false,
	},
	{
		Pattern: `**/bar/**/*`,
		Subject: `deep/foo/bar/`,
		Match:   true,
	},
	{
		Pattern: `**/bar/**/*`,
		Subject: `deep/foo/bar`,
		Match:   false,
	},
	{
		Pattern: `**/bar/**/*`,
		Subject: `deep/bar/bar`,
		Match:   false,
	}
```

Reply:
I've adopted these in, since I think they're very relevant to the code that's going in here, since we've fixed some other similar cases in this PR.